### PR TITLE
Add env for lux ad mosam

### PR DESCRIPTION
**Pull request by wilco375**
This PR adds an extra environment for Lux ad Mosam, which is deployed to https://luxstreep.csvalpha.nl / `/opt/docker/sofia/luxproduction`

### CODE REVIEW

**Review by Ellen-Wittingen — CHANGES_REQUESTED** [review]
Do we want a different image for Lux here? [https://github.com/csvalpha/sofia/blob/6ce23029f29c5f62a03eaee410e53d57843a83bb/app/views/invoices/show.html.erb#L7](https://github.com/csvalpha/sofia/blob/6ce23029f29c5f62a03eaee410e53d57843a83bb/app/views/invoices/show.html.erb#L7)

File: .github/workflows/continuous-delivery.yml [hunk 1]
```diff
@@ -20,11 +27,17 @@ jobs:
...
     steps:
       - name: Validate branch
+        env:
+          TARGET_ENV: ${{ github.event.inputs.targetenv }}
         run: |
           if [ "$GITHUB_REF_NAME" != 'staging' ] && [ "$GITHUB_REF_NAME" != 'master' ]; then
             echo 'This workflow can only be run on branches staging and master.'
             exit 1
           fi
+          if [ "$GITHUB_REF_NAME" != 'master' ] && [ "$TARGET_ENV" == 'luxadmosam' ]; then
```
Comment (guidojw) [inline .github/workflows/continuous-delivery.yml:37]: nit, would switch these for more clarity of intention

File: .github/workflows/continuous-delivery.yml [hunk 2]
```diff
@@ -7,6 +7,13 @@ on:
         description: Merge staging into master first? (y/N)
         required: false
         default: 'n'
+      targetenv:
```
Comment (guidojw) [inline .github/workflows/continuous-delivery.yml:10]: nit: I used snake_case at other places, can you do that here too?

File: .github/workflows/continuous-delivery.yml [hunk 3]
```diff
@@ -7,6 +7,13 @@ on:
...
+        type: choice
+        description: Target environment
+        options:
+        - csvalpha
+        - luxadmosam
```
Comment (guidojw) [inline .github/workflows/continuous-delivery.yml:15]:
```suggestion
          - csvalpha
          - luxadmosam
```

**Review by guidojw — COMMENTED** [review]
As requested a while ago, I checked the pipelines. Apart from Ellen's comment, it looks good!
only had some nits :)

File: app/controllers/credit_mutations_controller.rb [hunk 4]
```diff
@@ -17,7 +17,7 @@ def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
 
     respond_to do |format|
       if @mutation.save
-        NewCreditMutationNotificationJob.perform_later(@mutation) if Rails.env.production? || Rails.env.staging?
+        NewCreditMutationNotificationJob.perform_later(@mutation) if Rails.env.production? || Rails.env.staging? || Rails.env.luxproduction?
```
Comment (guidojw) [inline app/controllers/credit_mutations_controller.rb:20]: With this amount of environments, maybe here and on other places it's nicer to do something like `unless Rails.env.development?` (not sure if that's correct ruby)

**Comment by DrumsnChocolate** [comment]
what's the status of this? is this ready for merging and deploying?

**Comment by wilco375** [comment]
> Do we want a different image for Lux here?
> 
> https://github.com/csvalpha/sofia/blob/6ce23029f29c5f62a03eaee410e53d57843a83bb/app/views/invoices/show.html.erb#L7

Good point, for now I added a check that only shows an image for Flux.

**Comment by wilco375** [comment]
> what's the status of this? is this ready for merging and deploying?

Sorry for the late update, processed the feedback and should be ready to merge now. Can someone re-review it?

File: app/models/product.rb [hunk 5]
```diff
@@ -10,7 +11,7 @@ class Product < ApplicationRecord
   accepts_nested_attributes_for :product_prices, allow_destroy: true
 
   def requires_age
-    %w[beer craft_beer distilled wine tobacco].include? category
+    %w[beer craft_beer distilled whiskey wine tobacco].include? category
```
Comment (Ellen-Wittingen) [inline app/models/product.rb:14]: is low_alcohol_beer also age-restricted or can anyone drink it? (Not sure on Dutch law on this)

File: db/seeds/products.rb [hunk 6]
```diff
@@ -5,7 +5,8 @@ def seed_products # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metri
   products_beer = ['Bier (glas)', 'Bier (pul)', 'Bier (pitcher)', '12+1']
   products_craft_beer = ['Speciaalbier', 'Speciaalbier (pul)']
   products_non_alcoholic = ['Fris', 'Fris (klein)', 'Red Bull']
-  products_distilled = ['Sterke drank', 'Dure Whisky', 'Weduwe Joustra Beerenburg']
+  products_distilled = ['Sterke drank', 'Weduwe Joustra Beerenburg']
```
Comment (Ellen-Wittingen) [inline db/seeds/products.rb:8]: Do you want to add `low_alcohol_beer` here as well?

File: spec/factories/product.rb [hunk 7]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :product do
     name { Faker::Book.title }
-    category { %i[beer craft_beer non_alcoholic distilled wine food tobacco donation].sample }
+    category { %i[beer craft_beer non_alcoholic distilled whiskey wine food tobacco donation].sample }
```
Comment (Ellen-Wittingen) [inline spec/factories/product.rb:4]: Do you want to add `low_alcohol_beer` here as well?

File: spec/models/product_spec.rb [hunk 8]
```diff
@@ -27,7 +27,7 @@
 
   describe '#requires_age' do
     context 'when with requires age category' do
-      subject(:product) { create(:product, category: %w[beer craft_beer distilled wine tobacco].sample) }
+      subject(:product) { create(:product, category: %w[beer craft_beer distilled whiskey wine tobacco].sample) }
```
Comment (Ellen-Wittingen) [inline spec/models/product_spec.rb:30]: If  `low_alcohol_beer` is age-restricted, also put it here

**Review by Ellen-Wittingen — CHANGES_REQUESTED** [review]

**Comment by Ellen-Wittingen** [comment]
All seems good. Just had some comments about `low_alcohol_beer` since it was added after my last review

File: app/models/product.rb (hunk 5 above)
Comment (Ellen-Wittingen) [reply app/models/product.rb:14]: Like is it 2.0% Radler and such (probs age restricted?) or 0.0% Radler (is this age restricted?) and such, or something different?

Comment (wilco375) [reply app/models/product.rb:14]: It's a category for Lux, we won't use it in Flux. But I asked them and it's beer below 0.5% alcohol, which is not age restricted.

Comment (Ellen-Wittingen) [reply app/models/product.rb:14]: :+1:

**Review by Ellen-Wittingen — APPROVED** [review]

**Pull request merged**